GitCommands/RevisionReader.cs

```diff
@@ -468,45 +468,17 @@ int CountParents(in ReadOnlySpan<byte> array, int baseOffset)
                     currentOffset++;
                 }
 
-                // Removes empty Notes markers (this is the most common case)
-                bool hasNonEmptyNotes = _hasNotes;
-                if (hasNonEmptyNotes)
+                if (_hasNotes)
                 {
-                    if (decoded.EndsWith(_notesMarker))
+                    if (!decoded.EndsWith(_notesMarker) && ((ReadOnlySpan<char>)decoded).LastIndexOf(_notesMarker, StringComparison.Ordinal) is int notesStartIndex and >= 0)
                     {
-                        // Remove the empty marker
-                        decoded = decoded[..^_notesMarker.Length].TrimEnd();
-                        hasNonEmptyNotes = false;
+                        revision.Body = decoded[..notesStartIndex].TrimEnd().ToString();
```

Comment:
We have used [..n] instead of Slice(0,n) before fore readability.
(This is performance critical so Slice(n) is used)
https://github.com/gitextensions/gitextensions/pull/11417#discussion_r1416355271